sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -684,6 +685,24 @@ object TypeCoercion {
     }
   }
 
+  /**
+   * From SPARK-16323 division operator returns Long.
+   * So need to add the rule to cast to Long if there operands are of Byte, Short and Integer
```

Comment:
how about
```
`IntegralDivision` always returns long-type value. This rule cast the integral inputs to long type, to avoid
overflow during calculation.
```